src/main/java/org/sopt/app/application/calendar/CalendarServiceImpl.java

```diff
@@ -63,9 +63,14 @@ private List<Calendar> cacheAllCalendarResponse() {
     }
 
     private Optional<Calendar> getRecentCalendar(List<Calendar> calendars) {
-        return calendars.stream()
-                .filter(calendar -> !calendar.getStartDate().isBefore(CurrentDate.now))
-                .findFirst();
+        List<Calendar> sorted = calendars.stream()
+            .sorted(Comparator.comparing(Calendar::getStartDate)
+                .thenComparing(Calendar::getEndDate))
+            .toList();
+
+        return sorted.stream()
+            .filter(calendar -> !calendar.getStartDate().isBefore(CurrentDate.now()))
+            .findFirst();
```

Comment:
특별히 스트림을 분리한 이유가 있으실까요? 
없으시다면 가독성이나 성능 측면에서 하나의 스트림으로 처리하는 것이 좋다고 보여지는데, 의견이 궁금합니다!

```suggestion
return calendars.stream()
    .sorted(Comparator.comparing(Calendar::getStartDate)
        .thenComparing(Calendar::getEndDate))
    .filter(calendar -> !calendar.getStartDate().isBefore(CurrentDate.now()))
    .findFirst();
```

Reply:
기존에 로컬에서 로그 찍어보려고 분리했던걸 다시 안 합쳐뒀네요 합쳐두겠습니다!